clang/lib/CConv/Constraints.cpp

```diff
@@ -355,48 +380,42 @@ static std::set<VarAtom *> findBounded(ConstraintsGraph &CG,
 }
 
 bool Constraints::graph_based_solve(ConstraintSet &Conflicts) {
-  ConstraintsGraph ChkCG;
-  ConstraintsGraph PtrTypCG;
   std::set<Implies *> SavedImplies;
   std::set<Implies *> Empty;
   ConstraintsEnv &env = environment;
 
+  assert (ChkCG != nullptr && PtrTypCG != nullptr &&
+         "Constraints graphs cannot be nullptr");
+
   // Checked well-formedness
   environment.checkAssignment(getDefaultSolution());
 
   // Setup the Checked Constraint Graph.
   for (const auto &C : constraints) {
-    if (Geq *G = dyn_cast<Geq>(C)) {
-      if (G->constraintIsChecked())
-        ChkCG.addConstraint(G, *this);
-      else
-        PtrTypCG.addConstraint(G, *this);
-    }
-      // Save the implies to solve them later.
-    else if (Implies *Imp = dyn_cast<Implies>(C)) {
+    // Save the implies to solve them later.
+    if (Implies *Imp = dyn_cast<Implies>(C)) {
       assert(Imp->getConclusion()->constraintIsChecked() &&
           Imp->getPremise()->constraintIsChecked());
       SavedImplies.insert(Imp);
-    } else
-      llvm_unreachable("Bogus constraint type");
+    }
   }
 
   if (DebugSolver) {
     GraphVizOutputGraph::dumpConstraintGraphs(
-        "initial_constraints_graph.dot", ChkCG, PtrTypCG);
+        "initial_constraints_graph.dot", *ChkCG, *PtrTypCG);
   }
 
   // Solve Checked/unchecked constraints first
   env.doCheckedSolve(true);
-  bool res = do_solve(ChkCG, SavedImplies, env, this, true, nullptr, Conflicts);
+  bool res = do_solve(*ChkCG, SavedImplies, env, this, true, nullptr, Conflicts);
```

Comment:
What is the `*ChkCG` doing? Will that end up making a copy of the graph? Seems like doing so would be a bad idea?

Reply:
It is passed by reference, so there is no copy. Anyways, we changed the code so this doesn't happen anymore.